src/python_testing/matter_testing_support.py

```diff
@@ -1492,19 +1492,35 @@ def parse_matter_test_args(argv: Optional[List[str]] = None) -> MatterTestConfig
 
     return convert_args_to_matter_config(parser.parse_known_args(argv)[0])
 
+def async_to_sync(awaitable):
+    "Call sync, the async function."
+    # Usage of new implementation of ChipDeviceController cause the necessary to not use asyncio.run,
+    # because the initialization of `asyncio.Lock` use the `asyncio.get_event_loop()` on init.
+    # Usage of `asyncio.run()` and `asyncio.get_event_loop()` are exclusive.
+    loop = asyncio.get_event_loop()
+    return loop.run_until_complete(awaitable)
+
+def generate_random_nodeid(excluded_nodeid: typing.Optional[typing.List] = None) -> int:
+    if excluded_nodeid:
+        while True:
+            node_id=random.randint(0, 2^64-1)
+            if node_id not in excluded_nodeid:
+                return node_id
+    else:
+        return random.randint(0, 2^64-1)
```

Comment:
Refactor loop not to need a repeated random.randint by just generating until "not invalid".

Reply:
Done, used as recursive function